extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/execution/SelectorBasedPodTemplateSelectStrategy.java

```diff
@@ -64,9 +61,18 @@ public PodTemplate getPodTemplateForTask(Task task, Map<String, PodTemplate> tem
                                   .filter(selector -> selector.evaluate(task))
                                   .findFirst()
                                   .map(Selector::getSelectionKey)
-                                  .orElse(defaultKey);
+                                  .orElse("base");
 
-    return templates.getOrDefault(templateKey, templates.get("base"));
+    PodTemplate podTemplate = templates.getOrDefault(templateKey, templates.get("base"));
```

Comment:
A non-null `podTemplate` will always be returned in this line of code as when loading templates it will make sure `base` template get configured [here](https://github.com/apache/druid/blob/master/extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/taskadapter/PodTemplateTaskAdapter.java#L234)